common/include/fs/BDVirtualDevice.h

```diff
@@ -50,7 +50,7 @@ class BDVirtualDevice
      * @param buffer to save the data that has been read
      *
      */
-    virtual int32 readData(uint32 offset, uint32 size, char *buffer);
+    virtual __attribute__((warn_unused_result)) int32 readData(uint32 offset, uint32 size, char *buffer);
```

Comment:
Use [[[nodiscard]]](https://en.cppreference.com/w/cpp/language/attributes/nodiscard) instead of `__attribute__((warn_unused_result))`